plugins/Terminal/js/components/commandinput.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import { Map } from 'immutable'
+
+//These commands need a password prompt.
+const specialCommands = [ ['wallet', 'unlock'], ['wallet', 'load', 'seed'], ['help'], ['?'] ]
+
+const CommandInput = ({currentCommand, showCommandOverview, actions}) => {
+	componentDidUpdate: {
+		setTimeout(() => document.getElementById('command-input').focus(), 0)
+	}
+
+	reduce: {
+		const handleTextInput = (e) => {
+			actions.setCurrentCommand(e.target.value)
+		}
+
+		const handleKeyboardPress = (e) => {
+
+			//Enter button.
+			if (e.keyCode === 13) {
+
+				//Check if command is special.
+				switch (  isCommandSpecial(currentCommand, specialCommands) ) {
+				case -1: //Regular command.
```

Comment:
magic return values are bad. Instead of `isCommandSpecial` returning a number, maybe have a `commandType` function that returns a constant.

This may also not be the right place for your command-handling logic. Generally, components are only concerned with presentation. Their `onChange` functions are passed as props, unless they are very simple. @johnathanhowell can probably advise you better on this.